dataflow-migrate-schedules/README.adoc

```diff
@@ -0,0 +1,110 @@
+= Cloud Scheduler Migration
+
+The purpose of this project is to migrate existing schedules created with Spring
+Cloud Data Flow 2.2.x and before to the new 2.3.0 format and stage the
+SchedulerTaskLauncher.  This is a Spring Boot application that utilizes Spring Batch to create a workflow
+to migrate the schedules.  This is a single step Spring Batch Job that does the following:
+
+* Read - Retrieves all schedules from scheduler.
+
+* Process - Enriches the Schedule request with App and Deployer properties from the scheduler (or deployed app)
+as well as data from the TaskDefinition.
+
+* Write - Deploys artifacts if required and creates the new schedule.  Once the migrated
+schedule has been created, the old schedule is destroyed.
+
+In the case that the application fails with an exception.  You can re-execute the
+application and it will pick up where it left off.   Thanks Spring Batch! :-)
+
+== Build the project
+
+=== Services Required
+In order to migrate the existing schedules to the 2.3.x Spring Cloud Data Flow Scheduling format, the Schedule Migrator requires the following services:
+
+1. Access to the database where Spring Cloud Data Flow stores its Task Definitions.  For Cloud Foundry we need to bind the database to the Schedule Migrator.
+2. Access to the Scheduling Agent.  For Cloud Foundry we need to bind the PCF Scheduler to the Scheduler Migrator.
+
+=== Running the maven command
+
+```
+./mvnw clean package
+```
+
+== Running The Project For Cloud Foundry
+
+=== Launching your migration
+1) Create a manifest.yml file in a work directory.
+```
+---
+applications:
+- name: schedulemigrator
+  host: schedulemigrator
+  memory: 1G
+  disk_quota: 1G
+  instances: 0
+  path: <location of the jar>
+  env:
+    SPRING_APPLICATION_NAME: schedulemigrator
+    spring_cloud_deployer_cloudfoundry_url: <CF API URL>
+    spring_cloud_deployer_cloudfoundry_org: <Your org>
+    spring_cloud_deployer_cloudfoundry_space: <your space>
+    spring_cloud_deployer_cloudfoundry_username: <Your CF user name>
+    spring_cloud_deployer_cloudfoundry_password: <Your CF password>
+    spring_cloud_deployer_cloudfoundry_skipSslValidation: <true/false>
+    spring_cloud_deployer_cloudfoundry_services: <your SCDF database service,your scheduler service>
+    spring_cloud_scheduler_cloudfoundry_schedulerUrl: <URL to scheduler service>
+    spring_profiles_active: cf
+    spring.cloud.deployer.cloudfoundry.healthCheckTimeout: 300
+    spring.cloud.deployer.cloudfoundry.apiTimeout: 300
+    dataflowServerUri: <Your SCDF server URI>
```

Comment:
We need to be clear that this is the 2.3 instance and not the 2.2 instance.

Reply:
Gotcha.   Updated.